pilot/pkg/serviceregistry/serviceentry/servicediscovery.go

```diff
@@ -744,6 +734,14 @@ func servicesDiff(os []*model.Service, ns []*model.Service) ([]*model.Service, [
 	return added, deleted, updated, unchanged
 }
 
+func servicesEqual(os *model.Service, ns *model.Service) bool {
+	// Disabling `go vet` warning since this is actually safe in this case.
+	// nolint: vet
+	s := *os
+	s.AutoAllocatedAddress = ""
+	return reflect.DeepEqual(&s, ns)
```

Comment:
Is this going to compare things like mutex, etc? Seems odd... maybe mutex just is always considered equal though. I guess we already have it